index.js

```diff
@@ -13,133 +13,136 @@
+            );
+          }
+          break;
+        }
 
-                // Then, replace the whole path with an emotion one!
-                path.replaceWith(
-                    t.importDeclaration(
-                        [t.importDefaultSpecifier(t.identifier("styled"))],
-                        t.stringLiteral(opts.preact ? "preact-emotion" : "react-emotion")
-                    )
-                );
-            },
+        // replace <glamorous.Div/> with `<div/>`
+        case "JSXMemberExpression": {
+          const grandParent = path.parentPath.parent;
+          grandParent.name = t.identifier(grandParent.name.property.name.toLowerCase());
+          if (t.isJSXOpeningElement(grandParent) && grandParent.attributes) {
+            console.warn(
+              "The current version of the codemod has only weak support of the '<glamorous.Div>' syntax. It won't transform any attributes."
+            );
+          }
+          break;
+        }
 
-            /**
-             * Lastly, find all glamorous.Something() calls
-             * and replace them with emotion('something') calls.
-             */
-            CallExpression(path) {
+        default: {
+          console.warning("Found glamorous being used in an unkonwn context:", path.parent.type);
+        }
+      }
+    },
+  };
 
-                /**
-                 * First, rewrite all glamorous(Something) calls.
-                 */
-                if (
-                    t.isIdentifier(path.node.callee) &&
-                    path.node.callee.name === "glamorous"
-                ) {
-                    path.node.callee.name = "styled";
-                    return;
-                }
+  return {
+    name: "glamorousToEmotion",
+    visitor: {
+      ImportDeclaration(path, {opts}) {
+        const {value: libName} = path.node.source;
+        if (libName !== "glamorous" && libName !== "glamorous.macro") {
+          return;
+        }
+
+        // use "styled" as new default import, only if there's no such variable in use yet
+        const newName = path.scope.hasBinding("styled")
+          ? path.scope.generateUidIdentifier("styled").name
+          : "styled";
+        let newImports = [];
+        let useDefaultImport = false;
 
-                /**
-                 * Now, rewrite all glamorous.div(someStyle) calls.
-                 */
-                if (!t.isMemberExpression(path.node.callee)) {
-                    return;
-                }
-                if (path.node.callee.object.name !== "glamorous") {
-                    return;
-                }
-                path.replaceWith(
-                    t.callExpression(
-                        t.callExpression(t.identifier("styled"), [
-                            t.stringLiteral(path.node.callee.property.name)
-                        ]),
+        // only if the traversal below wants to know the newName,
+        // we're gonna add the default import
+        const getNewName = () => {
```

Comment:
How does this work on line 61 if it's defined here? 🤔 Surely I've missed something.

Reply:
Whatever you put as a second argument when calling a traverse step, will be available as `state` in the second parameter of each Visitor.

I pass the `getNewName` parameter [here](https://github.com/TejasQ/babel-plugin-glamorous-to-emotion/blob/127e82f102111122c3ebc685be389af73b5292ec/index.js#L127)

Check [this part of kentcdodds best practices](https://github.com/kentcdodds/babel-plugin-handbook/blob/master/README.md#optimizing-nested-visitors) for more information!

Reply:
Ah! I missed the part where it was being passed in on line 44! Makes sense!